src/operators.jl

```diff
@@ -496,13 +496,13 @@ end
 
 # Don't do size checks here in _return_array, defer that to (*)
 function _return_array{R,S}(A::AbstractArray{R}, x::AbstractArray{S,1})
-    Q = (R <: JuMPTypes && S <: JuMPTypes) ? QuadExpr : AffExpr
+    Q = promote_type(R,S)
```

Comment:
This works the same way on v0.3 and v0.4? If so that's great, I was never happy about this.